src/util.h

```diff
@@ -9,6 +9,13 @@
 #include <stdint.h>
 #include <stdio.h>
 
+/**
+ * \brief URL characters that can be escaped
+ */
+typedef enum  {
```

Comment:
Why don't we specify the actual list of characters that can be escaped here? This is very indirect.